lib/std/fs/Dir.zig

```diff
@@ -1565,6 +1566,7 @@ fn openDirFlagsZ(self: Dir, sub_path_c: [*:0]const u8, flags: posix.O) OpenError
         error.FileLocksNotSupported => unreachable, // locking folders is not supported
         error.WouldBlock => unreachable, // can't happen for directories
         error.FileBusy => unreachable, // can't happen for directories
+        error.ProcessNotFound => unreachable, // can't happen for directories
```

Comment:
What if I try to open `/proc/12345` (as opposed to a file under it)? Or a subdirectory somewhere under `/proc/12345/...`?

Reply:
When you try `const proc = try std.posix.open("/proc/29103", .{}, 0);` and the PID is unavailable, you get `.NOENT => return error.FileNotFound,`.

However, if the PID was open correctly and the PID disapears, and you try `const fd = try std.posix.openat(proc, "cpuset", .{}, 0);`, then you get `.SRCH => return error.ProcessNotFound,`

Reply:
But then it sounds like this shouldn't be `unreachable` unless I'm misunderstanding?

Reply:
I think 'unreachable' is the right choice here, because if the error is caught there, something else fundamentally wrong. Omitting it won't work; the compiler will complain about the dependencies. And the same goes for the other errors caught there with 'unreachable'. The idea wasn't mine; I just added it because it makes sense to me.

Reply:
I don't think that can be prevented. So the only option is to catch the error using `catch`. I just don't quite understand what you're getting at.

Reply:
I'm saying the `error.ProcessNotFound => unreachable` line here seems wrong and should be removed so that the error just goes to the `else => |e| return e` below it.